File: Services/Catalog/MC.Catalog.Domain/Entities/Product.cs
namespace MC.Catalog.Domain.Entities;

public class Product
{
    public string Id { get; set; } = string.Empty;
    public string OwnerId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public uint CategoryId { get; set; }
    public Category Category { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public List<string> ImageLinks { get; set; }
    public List<string> Tags { get; set; }
    public List<Tuple<string, string>> Parameters { get; set; }
}

File: Clients/mc.market.webapp/src/models/product.ts
export interface Product {
  id: string;
  title: string;
  description: string;
  price: number;
  imageLink: string;
  categoryName?: string;
  parameters?: ProductParameter[];
}

export interface Category {
  id: number;
  name: string;
}

export interface ProductParameter {
  name: string;
  value: string;
}

export interface ProductDetail {
  id: string;
  title: string;
  description: string;
  categoryId: number;
  category: Category;
  createdAt: string;
  price: number;
  stockQuantity: number;
  imageLinks: string[];
  tags: string[];
  parameters: ProductParameter[];
}

export interface ProductPaginatedResponse {
  items: Product[];
  pageSize: number;
  pageIndex: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}

export interface CreateProductPayload {
  ownerId: string;
  title: string;
  description: string;
  categoryId: number;
  price: number;
  stockQuantity: number;
  imageLinks: string[];
  tags: string[];
  parameters: [string, string][];
}

export interface ProductSearchCriteria {
  searchText?: string;
  categoryId?: number;
  parameters?: ProductParameter[];
}
